root/static/scripts/common/components/TagEditor.js

```diff
@@ -284,21 +281,20 @@ class TagEditor extends React.Component<TagEditorProps, TagEditorState> {
       };
 
       if (!this.state.positiveTagsOnly || isAlwaysVisible(t)) {
-        const genre = isGenre(t);
-
+        const isGenre = this.genreList.find(genre => (genre.name === t.tag.name));
```

Comment:
I noticed the primary use of `genreList` is searching the entire list for a specific tag name. It might be better then if it were `genreMap`, so you could simply lookup genres by name directly.

There is a case below that does `genreList.map(genre => genre.name)` to get a list of names, but that could be replaced with a call to `Object.keys(genreMap)`.